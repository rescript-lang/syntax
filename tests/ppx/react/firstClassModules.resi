@@jsxConfig({version: 3})

module Select: {
  module type T = {
    type key
    type t
  }

  @react.component
  let make: (
    ~model: module(T with type t = 'a and type key = 'key),
    ~selected: option<'key>,
    ~onChange: option<'key> => unit,
    ~items: array<'a>,
  ) => React.element
}

@@jsxConfig({version: 4, mode: "classic"})

module Select: {
  module type T = {
    type key
    type t
  }

  @react.component
  let make: (
    ~model: module(T with type t = 'a and type key = 'key),
    ~selected: option<'key>,
    ~onChange: option<'key> => unit,
    ~items: array<'a>,
  ) => React.element
}
