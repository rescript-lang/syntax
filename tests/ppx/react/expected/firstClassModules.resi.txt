@@jsxConfig({version: 3})

module Select: {
  module type T = {
    type key
    type t
  }

  @obj
  external makeProps: (
    ~model: module(T with type t = 'a and type key = 'key),
    ~selected: option<'key>,
    ~onChange: option<'key> => unit,
    ~items: array<'a>,
    ~key: string=?,
    unit,
  ) => {
    "model": module(T with type t = 'a and type key = 'key),
    "selected": option<'key>,
    "onChange": (option<'key> => unit),
    "items": array<'a>,
  } = ""
  let make: React.componentLike<
    {
      "model": module(T with type t = 'a and type key = 'key),
      "selected": option<'key>,
      "onChange": (option<'key> => unit),
      "items": array<'a>,
    },
    React.element,
  >
}

@@jsxConfig({version: 4, mode: "classic"})

module Select: {
  module type T = {
    type key
    type t
  }
  type props<'model, 'selected, 'onChange, 'items> = {
    model: 'model,
    selected: 'selected,
    onChange: 'onChange,
    items: 'items,
  }

  let make: React.componentLike<
    props<
      module(T with type t = 'a and type key = 'key),
      option<'key>,
      (option<'key> => unit),
      array<'a>,
    >,
    React.element,
  >
}
